spec/liquid_tags/golang_tag_spec.rb

```diff
@@ -0,0 +1,41 @@
+require "rails_helper"
+
+RSpec.describe GolangTag, type: :liquid_tag do
+  describe "#link" do
+    let(:valid_link) { "https://play.golang.org/p/HmnNoBf0p1z" }
+
+    def generate_tag(link)
+      Liquid::Template.register_tag("golang", GolangTag)
+      Liquid::Template.parse("{% golang #{link} %}")
+    end
+
+    it "accepts valid input" do
+      expect { generate_tag(valid_link) }.not_to raise_error
+    end
+
+    it "renders valid input" do
+      template = generate_tag(valid_link)
+      expected = 'src="true"'
+      expect(template.render).to include(expected)
+    end
+
+    it "accepts only Golang Playground links" do
+      badurl = "https://example.com"
+      expect do
+        generate_new_liquid(badurl)
+      end.to raise_error(StandardError)
```

Comment:
This spec is accidentally green. It only fails because `generate_new_liquid` doesn't exist, so the `StandardError` that gets caught here is due to a missing method:

```
[2] dev.to(#<RSpec::ExampleGroups::GolangTag::Link>)> generate_new_liquid(badurl)
NoMethodError: undefined method `generate_new_liquid' for #<RSpec::ExampleGroups::GolangTag::Link:0x000000011f122dc8>
```

When you change it to `generate_tag` no error will be raised and the spec will fail. Please apply the suggestion above and change the error to `RuntimeErrror`:

```suggestion
      end.to raise_error(RuntimeError)
```